CRM/Contribute/BAO/Product.php

```diff
@@ -36,72 +30,34 @@ public static function retrieve(&$params, &$defaults) {
     return $premium;
   }
 
-  /**
-   * @deprecated - this bypasses hooks.
-   * @param int $id
-   * @param bool $is_active
-   * @return bool
-   */
-  public static function setIsActive($id, $is_active) {
-    CRM_Core_Error::deprecatedFunctionWarning('writeRecord');
-    if (!$is_active) {
-      $dao = new CRM_Contribute_DAO_PremiumsProduct();
-      $dao->product_id = $id;
-      $dao->delete();
-    }
-    return CRM_Core_DAO::setFieldValue('CRM_Contribute_DAO_Product', $id, 'is_active', $is_active);
-  }
-
   /**
    * Add a premium product to the database, and return it.
    *
+   * @deprecated
    * @param array $params
    *   Update parameters.
    *
    * @return CRM_Contribute_DAO_Product
    */
   public static function create($params) {
```

Comment:
If this function is now unused in core, let's add noisy deprecation.

Reply:
done